### PR TITLE
feat Add prometheus retries

### PR DESCRIPTION
**Pull request by Tetrergeru**

### CODE REVIEW

File: metric_source/prometheus/fetch.go [hunk 1]
```diff
@@ -15,7 +15,28 @@ import (
...
+		if err == nil {
+			return res, nil
+		}
+
+		prometheus.logger.Warning().
+			Error(err).
+			Int("retries left", prometheus.config.Retries-i-1).
+			Msg("Failed to fetch prometheus target")
+
+		time.Sleep(prometheus.config.RetryTimeout)
```
Comment (kissken) [inline metric_source/prometheus/fetch.go:32]: why should we sleep if it is last available attempt?

Comment (Tetrergeru) [reply metric_source/prometheus/fetch.go]: Fixed

**Review by kissken — APPROVED** [review]

**Review by kissken — APPROVED** [review]

**Pull request merged**